src/EventListener/DoctrineSqlLogger.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Scoutapm\ScoutApmBundle\EventListener;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Logging\LoggerChain;
+use Doctrine\DBAL\Logging\SQLLogger;
+use Scoutapm\ScoutApmAgent;
+
+final class DoctrineSqlLogger implements SQLLogger
+{
+    /** @var ScoutApmAgent */
+    private $agent;
+
+    public function __construct(ScoutApmAgent $agent)
+    {
+        $this->agent = $agent;
+    }
+
+    public static function register(?Connection $connection, ScoutApmAgent $agent) : self
+    {
+        $scoutSqlLogger = new self($agent);
+
+        if ($connection === null) {
+            // Doctrine is not configured...
+            return $scoutSqlLogger;
+        }
+
+        $connectionConfiguration = $connection->getConfiguration();
+
+        $currentLogger = $connectionConfiguration->getSQLLogger();
+
+        if ($currentLogger === null) {
+            $connectionConfiguration->setSQLLogger($scoutSqlLogger);
+
+            return $scoutSqlLogger;
+        }
+
+        // @todo if already a Logger chain, just add scoutSqlLogger
```

Comment:
Resolve `@todo` - looks note, looks like `addLogger` is deprecated anyway, so I think the correct course of action is to make a new chain wrapping, as we have below already (so just remove the `@todo`)